vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthCaptchaFragment.kt

```diff
@@ -40,22 +47,71 @@ class FtueAuthCaptchaFragment @Inject constructor(
 ) : AbstractFtueAuthFragment<FragmentFtueLoginCaptchaBinding>() {
 
     private val params: FtueAuthCaptchaFragmentArgument by args()
+    private var webViewBinding: ViewStubWebviewBinding? = null
     private var isWebViewLoaded = false
 
     override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueLoginCaptchaBinding {
-        return FragmentFtueLoginCaptchaBinding.inflate(inflater, container, false)
+        return FragmentFtueLoginCaptchaBinding.inflate(inflater, container, false).also {
+            it.loginCaptchaWebViewStub.setOnInflateListener { _, inflated ->
+                webViewBinding = ViewStubWebviewBinding.bind(inflated)
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        inflateWebViewOrShowError()
+    }
+
+    private fun inflateWebViewOrShowError() {
+        views.loginCaptchaWebViewStub.inflateWebView(onError = {
+            MaterialAlertDialogBuilder(requireActivity())
+                    .setTitle(R.string.dialog_title_error)
+                    .setMessage(it.localizedMessage)
+                    .setPositiveButton(R.string.ok) { _, _ ->
+                        requireActivity().recreate()
+                    }
+                    .show()
+        })
     }
 
     override fun resetViewModel() {
         viewModel.handle(OnboardingAction.ResetAuthenticationAttempt)
     }
 
     override fun updateWithState(state: OnboardingViewState) {
-        if (!isWebViewLoaded) {
-            captchaWebview.setupWebView(this, views.loginCaptchaWevView, views.loginCaptchaProgress, params.siteKey, state) {
+        if (!isWebViewLoaded && webViewBinding != null) {
+            captchaWebview.setupWebView(this, webViewBinding!!.root, views.loginCaptchaProgress, params.siteKey, state) {
                 viewModel.handle(OnboardingAction.PostRegisterAction(RegisterAction.CaptchaDone(it)))
             }
             isWebViewLoaded = true
         }
     }
 }
+
+private fun ViewStub.inflateWebView(onError: (Throwable) -> Unit) {
+    try {
+        inflate()
+    } catch (e: Exception) {
+        val isMissingWebView = e.crawlCausesFor { it.message?.contains("MissingWebViewPackageException").orFalse() }
```

Comment:
we have to crawl through the exception causes to find the missing webview package exception, it's a bit hacky~